packages/app/src/elements/import-dialog.ts

```diff
@@ -119,7 +288,9 @@ Github,"work,coding",https://github.com,john.doe@gmail.com,129lskdf93`)
                 this._items = await imp.asLastPass(file);
                 break;
             case imp.CSV.value:
-                this._items = await imp.asCSV(file);
+                const result = await imp.asCSV(file, this._itemColumns, this._csvHasDataOnFirstRowCheckbox.checked);
+                this._items = result.items;
+                this._itemColumns = result.itemColumns;
```

Comment:
This seems kind of circular. Are the `itemColumns` a result or an input of this function? I think it should be either/or. Better to have a separate function that infers the initial item column definitions and another that actually parses the data into vault items (taking the column definitions as an input). So something like this:

```ts
function inferCSVColumns(file: File, firstRowHasData: boolean): CSVItemColumn[] {}
function asCSV(file: File, itemColumns: CSVItemColumn[], firstRowHasData: boolean): VaultItem[] {}
```
Or, if you don't want to read/parse the file twice:

```ts
function inferCSVColumns(data: string[][], firstRowHasData: boolean): CSVItemColumn[] {}
function asCSV(data: string[][], itemColumns: CSVItemColumn[], firstRowHasData: boolean): VaultItem[] {}
```

Reply:
Actually I just realised my second definition for `asCSV` is exactly what `asTable` is. Well in any case I think there should be a clearer separation of concerns with these functions, but you don't have to untangle this right now if you don't want to.

Reply:
I don't feel strongly about it right now. I do like reusing it like this: if there's something set, use it, otherwise, parse and send it back. Anyway, as this comes closer to being done, we can see what makes sense to untangle or not. I don't want to refactor unnecessarily until this is all working as expected.